Mvn clean install has error msg.

```
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.039 s - in org.apache.shardingsphere.data.pipeline.core.task.IncrementalTaskTest
[INFO] Running org.apache.shardingsphere.data.pipeline.core.job.progress.JobProgressTest
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.021 s - in org.apache.shardingsphere.data.pipeline.core.job.progress.JobProgressTest
[INFO] Running org.apache.shardingsphere.data.pipeline.core.job.progress.yaml.JobProgressYamlSwapperTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.004 s - in org.apache.shardingsphere.data.pipeline.core.job.progress.yaml.JobProgressYamlSwapperTest
[INFO] Running org.apache.shardingsphere.data.pipeline.core.job.RuleAlteredJobIdTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.apache.shardingsphere.data.pipeline.core.job.RuleAlteredJobIdTest
[INFO] Running org.apache.shardingsphere.data.pipeline.core.job.FinishedCheckJobTest
[WARN ] 2022-03-25 10:48:50.601 [main] o.a.s.d.p.c.a.i.RuleAlteredJobAPIImpl - jobId already exists in registry center, ignore, jobConfigKey=/scaling/0130317c30317c3054317c6c6f6769635f6462/config
[ERROR] 2022-03-25 10:48:50.612 [main] o.a.s.d.p.core.job.FinishedCheckJob - scaling job 0130317c30317c3054317c6c6f6769635f6462 finish check failed!
java.lang.RuntimeException: Not necessary to restore source writing
        at org.apache.shardingsphere.data.pipeline.core.api.impl.RuleAlteredJobAPIImpl.restoreClusterWriteDB(RuleAlteredJobAPIImpl.java:198)
        at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.spi.DefaultSourceWritingStopAlgorithm.releaseLock(DefaultSourceWritingStopAlgorithm.java:42)
        at org.apache.shardingsphere.data.pipeline.core.job.FinishedCheckJob.execute(FinishedCheckJob.java:89)
        at org.apache.shardingsphere.data.pipeline.core.job.FinishedCheckJobTest.assertExecuteActiveJob(FinishedCheckJobTest.java:77)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
        at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
        at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
        at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:383)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:344)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:125)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:417)

```

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-memory-mode/shardingsphere-memory-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/memory/lock/ShardingSphereNonReentrantLock.java
@@ -46,9 +46,13 @@ public final class ShardingSphereNonReentrantLock implements ShardingSphereLock 
     }
     
     private synchronized boolean innerTryLock(final long timeout) {
+        if (locked) {
+            return false;
+        }
         try {
             if (innerLock.tryLock(timeout, TimeUnit.MILLISECONDS)) {
                 locked = true;
+                return true;
             }
             return false;
         } catch (final InterruptedException ignored) {
@@ -57,9 +61,11 @@ public final class ShardingSphereNonReentrantLock implements ShardingSphereLock 
     }
     
     @Override
-    public void releaseLock(final String lockName) {
-        innerLock.unlock();
-        locked = false;
+    public synchronized void releaseLock(final String lockName) {
+        if (locked) {
+            innerLock.unlock();
+            locked = false;
+        }
     }
     
     @Override

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/standalone/lock/ShardingSphereNonReentrantLock.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/standalone/lock/ShardingSphereNonReentrantLock.java
@@ -46,9 +46,13 @@ public final class ShardingSphereNonReentrantLock implements ShardingSphereLock 
     }
     
     private synchronized boolean innerTryLock(final long timeout) {
+        if (locked) {
+            return false;
+        }
         try {
             if (innerLock.tryLock(timeout, TimeUnit.MILLISECONDS)) {
                 locked = true;
+                return true;
             }
             return false;
         } catch (final InterruptedException ignored) {
@@ -58,8 +62,10 @@ public final class ShardingSphereNonReentrantLock implements ShardingSphereLock 
     
     @Override
     public void releaseLock(final String lockName) {
-        innerLock.unlock();
-        locked = false;
+        if (locked) {
+            innerLock.unlock();
+            locked = false;
+        }
     }
     
     @Override

